velox/exec/Task.cpp

```diff
@@ -418,6 +422,48 @@ void Task::removeSpillDirectoryIfExists() {
   }
 }
 
+/// Builds the query trace config.
+std::optional<QueryTraceConfig> Task::makeQueryTraceConfig() const {
+  const auto& queryConfig = queryCtx_->queryConfig();
+  const auto queryTraceDir = queryConfig.queryTraceDir();
+  if (queryTraceDir.empty()) {
```

Comment:
Shall we just use VELOX_CHECK here as we might not want to run the query if tracing is not setup properly? This is only used for the performance testing purpose?

Reply:
Fixed